### DISCUSSION TITLE
input/textarea组件关于maxlength的限制不能输入是否增加api

### DISCUSSION THREAD

**Discussion by carolin913** [Tencent/tdesign#123] [Ideas]
![wecom-temp-249ee0b56bfb526f4ef6e0e3b63837e6](https://user-images.githubusercontent.com/2638568/169285801-3ed4bef0-93ae-4105-b45b-5bbb77aeddb4.png)
如图，目前input和textarea组件中有maxlength的设置就会阻止输入，对于一些copy内容来发表，如微博发表等场景，使用时还需要先去其他地方编辑好，数好字数再贴进来，体验不是很友好。

参考了下arco
https://arco.design/react/components/input#%E5%AD%97%E6%95%B0%E7%BB%9F%E8%AE%A1
![wecom-temp-0a9b80c26be7639fb377e0a342ecd211](https://user-images.githubusercontent.com/2638568/169285744-f684bafa-157c-4ddc-bd0f-4ae333fb32a3.png)
感觉是比较合理的方案，大家可以看下是否可以如此扩展maxlength

**Comment by carolin913** [comment]
还有maxcharacter这种，如果都按erroronly这种方式扩展，会很冗余，建议还是使用一个新的api比较好

**Comment by carolin913** [comment]
讨论结果，采用allowInputOverMax

**Comment by webpon** [comment]
微信小程序的textarea同样遇到这个问题

**Reply by LeeJim** [reply]
后续会给小程序加上
